src/js/services/app.js

```diff
@@ -74,24 +75,41 @@ export function showChannelPage(channelType) {
         const {user, app: {integrations}} = getState();
         const channelDetails = CHANNEL_DETAILS[channelType];
         const isLinked = isChannelLinked(user.clients, channelType);
-        const openLink = channelDetails.getURL && (!channelDetails.Component || isLinked);
 
-        if (openLink) {
+        function _showChannelPage() {
+            dispatch(AppStateActions.showChannelPage(channelType));
+
+            return dispatch(connectToFayeUser())
+                .then(() => dispatch(channelDetails.onChannelPage()));
+        }
+
+        function openLink() {
             const appChannel = getIntegration(integrations, channelType);
-            const link = channelDetails.getURL(user, appChannel, isLinked);
+            const getCode = channelDetails.usesCode ?
```

Comment:
What's the behavior here? I'm surprised it doesn't trigger the browser popup blocker with all of this since we are calling `window.open` so late in the process. Otherwise, do we have some sort of ui feedback while it waits for a code?